test/test_qml.cpp

```diff
@@ -192,44 +164,16 @@ class Setup : public QObject
 
 
-      qmlRegisterType<QgsSnappingUtils>( "org.qgis", 1, 0, "SnappingUtils" );
-      qmlRegisterType<QgsMapLayerProxyModel>( "org.qgis", 1, 0, "MapLayerModel" );
-      qmlRegisterType<QgsVectorLayer>( "org.qgis", 1, 0, "VectorLayer" );
-      qmlRegisterType<QgsMapThemeCollection>( "org.qgis", 1, 0, "MapThemeCollection" );
-      qmlRegisterType<QgsLocatorProxyModel>( "org.qgis", 1, 0, "QgsLocatorProxyModel" );
-      qmlRegisterType<QgsVectorLayerEditBuffer>( "org.qgis", 1, 0, "QgsVectorLayerEditBuffer" );
-      qmlRegisterUncreatableType<Qgis>( "org.qgis", 1, 0, "Qgis", "" );
-      qmlRegisterUncreatableType<QgsProject>( "org.qgis", 1, 0, "Project", "" );
-      qmlRegisterUncreatableType<QgsProjectDisplaySettings>( "org.qgis", 1, 0, "ProjectDisplaySettings", "" );
-      qmlRegisterUncreatableType<QgsCoordinateReferenceSystem>( "org.qgis", 1, 0, "CoordinateReferenceSystem", "" );
-      qmlRegisterUncreatableType<QgsUnitTypes>( "org.qgis", 1, 0, "QgsUnitTypes", "" );
-      qmlRegisterUncreatableType<QgsRelationManager>( "org.qgis", 1, 0, "RelationManager", "The relation manager is available from the QgsProject. Try `qgisProject.relationManager`" );
-      qmlRegisterUncreatableType<QgsWkbTypes>( "org.qgis", 1, 0, "QgsWkbTypes", "" );
-      qmlRegisterUncreatableType<QgsMapLayer>( "org.qgis", 1, 0, "MapLayer", "" );
-      qmlRegisterUncreatableType<QgsVectorLayer>( "org.qgis", 1, 0, "VectorLayerStatic", "" );
-      qmlRegisterType<QgsQuickMapCanvasMap>( "org.qgis", 1, 0, "MapCanvasMap" );
-      qmlRegisterType<QgsQuickMapSettings>( "org.qgis", 1, 0, "MapSettings" );
-      qmlRegisterType<QgsQuickCoordinateTransformer>( "org.qfield", 1, 0, "CoordinateTransformer" );
-      qmlRegisterType<QgsQuickElevationProfileCanvas>( "org.qgis", 1, 0, "ElevationProfileCanvas" );
-      qmlRegisterType<QgsQuickMapTransform>( "org.qgis", 1, 0, "MapTransform" );
-
-
-      qRegisterMetaType<QgsGeometry>( "QgsGeometry" );
-      qRegisterMetaType<QgsFeature>( "QgsFeature" );
-      qRegisterMetaType<QgsPoint>( "QgsPoint" );
-      qRegisterMetaType<QgsPointXY>( "QgsPointXY" );
-
-      qmlRegisterType<ValueMapModel>( "org.qfield", 1, 0, "ValueMapModel" );
-      qmlRegisterType<QgsQuickCoordinateTransformer>( "org.qfield", 1, 0, "CoordinateTransformer" );
-      qmlRegisterUncreatableType<QAbstractSocket>( "org.qfield", 1, 0, "QAbstractSocket", "" );
-      qmlRegisterUncreatableType<AbstractGnssReceiver>( "org.qfield", 1, 0, "AbstractGnssReceiver", "" );
-      qRegisterMetaType<GnssPositionInformation>( "GnssPositionInformation" );
-      qmlRegisterType<Positioning>( "org.qfield", 1, 0, "Positioning" );
-
-      REGISTER_SINGLETON( "org.qfield", CoordinateReferenceSystemUtils, "CoordinateReferenceSystemUtils" );
+      QgsApplication *mApp;
+      QgisMobileapp::initDeclarative( mApp, engine );
+
+      QString mPath = QCoreApplication::applicationDirPath() + "/../../../resources/sample_projects/bees.qgz";
```

Comment:
We need to change that to rely on a project file that lives in test/testdata. I would just go for a single layer project with the bees geopackage.

Reply:
can you help me with this

Reply:
You can inspire yourself from this bit of code:

```
        if ( arguments[i] == QStringLiteral( "-input" ) )
        {
          if ( i + 1 < arguments.size() )
          {
            // the nmea server python script, relative to the absolute input path
            nmeaServerLocation = QString( "%1/../nmea_server" ).arg( arguments[i + 1] );
            mDataDir = QString( "%1/../testdata" ).arg( arguments[i + 1] );
          }
        }
```

In this case, you'd replace %1/../nmea_server with %1/../testdata/your_test_project.qgz where your_test_project is a simplified version of the sample bees project with a single apiary layer.

Reply:
may i use `mDataDir` ? its pointing at `testdata` dir.

Reply:
@mohsenD98 , true true, you could definitively use that.